# Analyse des dataset de la CALL

Communauté d'Aglomeration Lens Lievin<br>

Les fichiers proviennent du site : https://opendata.agglo-lenslievin.fr/ <br>
Le but de ce notebook est d'analyser et de préparer les fichier de la CALL.<br>

J'ai plusieurs fichiers :<br>
_Données essentielle de la [commande publique](#cmd_pub)<br>
_donnée INSEE sur la [population](#population)<br>
_liste des benne de [verre](#verre)<br>
_liste des établissement [scolaire](#ecole)<br>
_liste des établissement [numérique](#numerique)<br>


In [1]:
import pandas as pd
import geopandas as gpd

import fonctions.perso_pandas as perso_pd

<a id=cmd_pub></a>

## Fichier commande publique

In [2]:
data_commande_publique = pd.read_csv("CALL-data/Données_essentielles_de_la_commande_publique.csv")


In [3]:
perso_pd.analyseDataFrame(data_commande_publique)

################################# INFO  #################################
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 397 entries, 0 to 396
Data columns (total 78 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   X           393 non-null    float64
 1   Y           393 non-null    float64
 2   FID         397 non-null    int64  
 3   id          397 non-null    object 
 4   idmarche    397 non-null    int64  
 5   source      397 non-null    object 
 6   type        397 non-null    object 
 7   natureobje  397 non-null    object 
 8   objetmarch  397 non-null    object 
 9   codecpv_or  397 non-null    int64  
 10  codecpv     397 non-null    object 
 11  codecpv_di  397 non-null    int64  
 12  referencec  397 non-null    object 
 13  datenotifi  397 non-null    object 
 14  anneenotif  355 non-null    float64
 15  moisnotifi  397 non-null    object 
 16  datepublic  397 non-null    object 
 17  dureemois   397 non-null    int64  
 

X             Y         FID                id  \
count   3.930000e+02  3.930000e+02  397.000000               397   
unique           NaN           NaN         NaN               397   
top              NaN           NaN         NaN  2018ScG_m3Xgl300   
freq             NaN           NaN         NaN                 1   
mean    2.821299e+00  5.043726e+01  199.000000               NaN   
std     7.559140e-15  4.980139e-14  114.748275               NaN   
min     2.821299e+00  5.043726e+01    1.000000               NaN   
25%     2.821299e+00  5.043726e+01  100.000000               NaN   
50%     2.821299e+00  5.043726e+01  199.000000               NaN   
75%     2.821299e+00  5.043726e+01  298.000000               NaN   
max     2.821299e+00  5.043726e+01  397.000000               NaN   

             idmarche    source    type natureobje  \
count      397.000000       397     397        397   
unique            NaN         1       2          3   
top               NaN  API AIFE  Marché   Services   
freq              NaN       397     396        212   
mean    367560.725441       NaN     NaN        NaN   
std        115.094044       NaN     NaN        NaN   
min     367362.000000       NaN     NaN        NaN   
25%     367461.000000       NaN     NaN        NaN   
50%     367561.000000       NaN     NaN        NaN   
75%     367660.000000       NaN     NaN        NaN   
max     367759.000000       NaN     NaN        NaN   

                                               objetmarch    codecpv_or  ...  \
count                                                 397  3.970000e+02  ...   
unique                                                385           NaN  ...   
top     MarchXXXXX d'opXXXXXrateurs de tXXXXXlXXXXXcom...           NaN  ...   
freq                                                    5           NaN  ...   
mean                                                  NaN  5.711059e+07  ...   
std                                                   NaN  2.089442e+07  ...   
min                                                   NaN  0.000000e+00  ...   
25%                                                   NaN  4.511271e+07  ...   
50%                                                   NaN  5.053000e+07  ...   
75%                                                   NaN  7.163130e+07  ...   
max                                                   NaN  9.262000e+07  ...   

        distanceac  typeiden_1 id_cotitul denomina_1  typeiden_2 id_cotit_1  \
count   397.000000         397        397        397         397        397   
unique         NaN           2         52         51           2         38   
top            NaN                                                            
freq           NaN         337        337        337         357        357   
mean    119.992872         NaN        NaN        NaN         NaN        NaN   
std     181.605483         NaN        NaN        NaN         NaN        NaN   
min       0.000000         NaN        NaN        NaN         NaN        NaN   
25%      15.951517         NaN        NaN        NaN         NaN        NaN   
50%      37.203376         NaN        NaN        NaN         NaN        NaN   
75%     182.150634         NaN        NaN        NaN         NaN        NaN   
max     907.741660         NaN        NaN        NaN         NaN        NaN   

       denomina_2  typeiden_3 id_cotit_2  denomina_3  
count         397         397        397         397  
unique         38           2          8           8  
top                                                   
freq          357         390        390         390  
mean          NaN         NaN        NaN         NaN  
std           NaN         NaN        NaN         NaN  
min           NaN         NaN        NaN         NaN  
25%           NaN         NaN        NaN         NaN  
50%           NaN         NaN        NaN         NaN  
75%           NaN         NaN        NaN         NaN  
max           NaN         NaN        N

################################# HEAD  #################################


X          Y  FID                id  idmarche    source    type  \
0  2.821299  50.437255    1  2018ScG_m3Xgl300    367390  API AIFE  Marché   
1  2.821299  50.437255    2  2019lf5cgGvoNn00    367492  API AIFE  Marché   
2  2.821299  50.437255    3  20182M_y8DjwM200    367367  API AIFE  Marché   
3  2.821299  50.437255    4  2019_ejVJOvVCj00    367476  API AIFE  Marché   
4  2.821299  50.437255    5  2018nRq9mPFTtj00    367418  API AIFE  Marché   

    natureobje                                         objetmarch  codecpv_or  \
0  Fournitures  Fourniture de boissons pour les services de la...    15900000   
1     Services  19S143-Edition de brochures pour le Pays d'Art...    79822500   
2  Fournitures    Travaux d'eau potable rue de Gouy XXXXX SERVINS    44162500   
3      Travaux  19S158-Electricité courants forts et courants ...    45311200   
4     Services  Assistance XXXXX maXXXXXtrise d'ouvrage relati...    71330000   

   ...  distanceac  typeiden_1 id_cotitul denomina_1  typeiden_2 id_cotit_1  \
0  ...   59.131984                                                            
1  ...    5.540897                                                            
2  ...   48.644099                                                            
3  ...   24.606583                                                            
4  ...  185.986283                                                            

  denomina_2  typeiden_3  id_cotit_2  denomina_3  
0                                                 
1                                                 
2                                                 
3                                                 
4                                                 

[5 rows x 78 columns]

################################# DOUBLONS  #################################


Empty DataFrame
Columns: [X, Y, FID, id, idmarche, source, type, natureobje, objetmarch, codecpv_or, codecpv, codecpv_di, referencec, datenotifi, anneenotif, moisnotifi, datepublic, dureemois, dureemoise, dureemoisc, montant, nombretitu, montantcal, formeprix, lieuexecut, lieuexec_1, lieuexec_2, codedepart, coderegion, libellereg, nature, accord_cad, procedure_, idacheteur, sirenachet, nomacheteu, coderegi_1, libeller_1, departemen, libelledep, codepostal, libellearr, codearrond, libellecom, codecommun, superficie, population, typeidenti, siretetabl, sireteta_1, sirenetabl, nicetablis, sireneta_1, categoriee, denominati, coderegi_2, libeller_2, libelled_1, departem_1, libellea_1, codearro_1, codepost_1, adresseeta, communeeta, codecomm_1, codetypeet, superfic_1, populati_1, distanceac, typeiden_1, id_cotitul, denomina_1, typeiden_2, id_cotit_1, denomina_2, typeiden_3, id_cotit_2, denomina_3]
Index: []

[0 rows x 78 columns]

################################## recherche de clé pour merge ###################
la colonne FID est une clé
la colonne id est une clé
la colonne idmarche est une clé


Les 3 identifiants sont des clés (Normal) <br>
Il n y a pas de doublons.<br>
397 lignes et 78 colonnes.<br>
Colonne à enlevé :<br>
X,Y, 
id , idmarché, 
codecpv_or, codecpv, 
datenotifi, anneenotif,moisnotifi,dureemoise,dureemoisc,montant,nombretitu,
lieuexec_1,lieuexec_2,coderegion,libellereg,accord_cad,
idacheteur,sirenachet,nomacheteu,coderegi_1,libeller_1,departemen,libelledep,codepostal,libellearr,
codearrond,libellecom,codecommun,superficie,population,typeidenti,sirettabl,sireteat_1,
sirenetabl,nicetablis,sereneta_1,coderegi_2,libeller_2,departem_1,libellea_1,codearro_1,codepost_1,
adresseta,communeeta,codecomm_1,codetypet,superfic_1,populati_1,distanceac,typeeiden_1,id_cotitul,
denomina_1,typeiden_2,id_cotit_1,denomina_2,typeiden_3,id_cotit_2,denomina_3
<br><br>
Changer le nom de colonnes pour être plus intuitif<br>




In [4]:
#enlever les colonnes inutiles
cmd_pub =data_commande_publique.drop(['id','idmarche','codecpv_or','codecpv','datenotifi',
                             'anneenotif','moisnotifi','dureemoise','dureemoisc','montant',
                             'nombretitu','lieuexec_1','lieuexec_2','coderegion','libellereg',
                             'accord_cad','idacheteur','sirenachet','nomacheteu','coderegi_1','libeller_1',
                             'departemen','libelledep','codepostal','libellearr','codearrond',
                             'libellecom','codecommun','superficie','population','typeidenti','siretetabl',
                             'sireteta_1','sirenetabl','nicetablis','sireneta_1','coderegi_2',
                             'libeller_2','departem_1','libellea_1','codearro_1','codepost_1',
                             'adresseeta','communeeta','codecomm_1','codetypeet','superfic_1',
                             'populati_1','distanceac','typeiden_1','id_cotitul','denomina_1','lieuexecut',
                             'typeiden_2','id_cotit_1','denomina_2','typeiden_3','id_cotit_2','denomina_3'
                                      #,'FID'
                            ], axis=1)

#changer le nom de certaine colonnes
cmd_pub.rename(columns = {'objetmarch':'description','montantcal':'montant',
                         'libelled_1':'departement entreprise',
                         'codedepart':'departement executant'}, inplace = True)

commande publique final :

In [5]:
perso_pd.analyseDataFrame(cmd_pub)

################################# INFO  #################################
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 397 entries, 0 to 396
Data columns (total 19 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   X                       393 non-null    float64
 1   Y                       393 non-null    float64
 2   FID                     397 non-null    int64  
 3   source                  397 non-null    object 
 4   type                    397 non-null    object 
 5   natureobje              397 non-null    object 
 6   description             397 non-null    object 
 7   codecpv_di              397 non-null    int64  
 8   referencec              397 non-null    object 
 9   datepublic              397 non-null    object 
 10  dureemois               397 non-null    int64  
 11  montant                 397 non-null    int64  
 12  formeprix               397 non-null    object 
 13  departement executant

X             Y         FID    source    type natureobje  \
count   3.930000e+02  3.930000e+02  397.000000       397     397        397   
unique           NaN           NaN         NaN         1       2          3   
top              NaN           NaN         NaN  API AIFE  Marché   Services   
freq             NaN           NaN         NaN       397     396        212   
mean    2.821299e+00  5.043726e+01  199.000000       NaN     NaN        NaN   
std     7.559140e-15  4.980139e-14  114.748275       NaN     NaN        NaN   
min     2.821299e+00  5.043726e+01    1.000000       NaN     NaN        NaN   
25%     2.821299e+00  5.043726e+01  100.000000       NaN     NaN        NaN   
50%     2.821299e+00  5.043726e+01  199.000000       NaN     NaN        NaN   
75%     2.821299e+00  5.043726e+01  298.000000       NaN     NaN        NaN   
max     2.821299e+00  5.043726e+01  397.000000       NaN     NaN        NaN   

                                              description  codecpv_di  \
count                                                 397  397.000000   
unique                                                385         NaN   
top     MarchXXXXX d'opXXXXXrateurs de tXXXXXlXXXXXcom...         NaN   
freq                                                    5         NaN   
mean                                                  NaN   57.697733   
std                                                   NaN   20.026102   
min                                                   NaN    0.000000   
25%                                                   NaN   45.000000   
50%                                                   NaN   50.000000   
75%                                                   NaN   72.000000   
max                                                   NaN   92.000000   

               referencec  datepublic   dureemois       montant  formeprix  \
count                 397         397  397.000000  3.970000e+02        397   
unique                126         148         NaN           NaN          4   
top     Services d'études  2021/08/24         NaN           NaN  Révisable   
freq                   34          20         NaN           NaN        280   
mean                  NaN         NaN   26.060453  5.881230e+05        NaN   
std                   NaN         NaN   21.811900  4.812090e+06        NaN   
min                   NaN         NaN    0.000000  0.000000e+00        NaN   
25%                   NaN         NaN   12.000000  2.677500e+04        NaN   
50%                   NaN         NaN   18.000000  8.000000e+04        NaN   
75%                   NaN         NaN   36.000000  2.839000e+05        NaN   
max                   NaN         NaN  180.000000  9.050000e+07        NaN   

        departement executant  nature         procedure_ categoriee  \
count              397.000000     397                397        397   
unique                    NaN       4                  5          4   
top                       NaN  MARCHE  Procédure adaptée        PME   
freq                      NaN     227                196        183   
mean                61.833753     NaN                NaN        NaN   
std                  4.946948     NaN                NaN        NaN   
min                  0.000000     NaN                NaN        NaN   
25%                 62.000000     NaN                NaN        NaN   
50%                 62.000000     NaN                NaN        NaN   
75%                 62.000000     NaN                NaN        NaN   
max                 92.000000     NaN                NaN        NaN   

       denominati departement entreprise  
count         397                    397  
unique        258                     33  
top     SADE CGTH          Pas-de-Calais  
freq           11                    165  
mean          NaN                    NaN  
std           NaN                    NaN  
min           NaN                    NaN  
25%           NaN                    NaN  
50%           Na

################################# HEAD  #################################


X          Y  FID    source    type   natureobje  \
0  2.821299  50.437255    1  API AIFE  Marché  Fournitures   
1  2.821299  50.437255    2  API AIFE  Marché     Services   
2  2.821299  50.437255    3  API AIFE  Marché  Fournitures   
3  2.821299  50.437255    4  API AIFE  Marché      Travaux   
4  2.821299  50.437255    5  API AIFE  Marché     Services   

                                         description  codecpv_di  \
0  Fourniture de boissons pour les services de la...          15   
1  19S143-Edition de brochures pour le Pays d'Art...          79   
2    Travaux d'eau potable rue de Gouy XXXXX SERVINS          44   
3  19S158-Electricité courants forts et courants ...          45   
4  Assistance XXXXX maXXXXXtrise d'ouvrage relati...          71   

                             referencec  datepublic  dureemois  montant  \
0  Boissons, tabac et produits connexes  2020/03/27         12     5319   
1      Services de conception graphique  2022/12/20          3     5711   
2               Conduites d'eau potable  2020/04/06          2   127990   
3   Travaux d'installations électriques  2020/08/11         16    49861   
4          Services divers d'ingénierie  2020/04/05         48    56945   

               formeprix  departement executant        nature  \
0              Révisable                     62  ACCORD-CADRE   
1                  Ferme                     62        MARCHE   
2                  Ferme                     62        MARCHE   
3  Ferme et actualisable                     62        MARCHE   
4              Révisable                     62        MARCHE   

              procedure_ categoriee              denominati  \
0      Procédure adaptée         NC                 Bedague   
1      Procédure adaptée        PME            L'ARTESIENNE   
2      Procédure adaptée         GE  SOGEA NORD HYDRAULIQUE   
3      Procédure adaptée         NC                 SATELEC   
4  Appel d'offres ouvert         GE                  PARVIS   

  departement entreprise  
0          Pas-de-Calais  
1          Pas-de-Calais  
2                   Nord  
3                   Nord  
4                  Paris

################################# DOUBLONS  #################################


Empty DataFrame
Columns: [X, Y, FID, source, type, natureobje, description, codecpv_di, referencec, datepublic, dureemois, montant, formeprix, departement executant, nature, procedure_, categoriee, denominati, departement entreprise]
Index: []

################################## recherche de clé pour merge ###################
la colonne FID est une clé


In [6]:
# création du fichier commande publique final
cmd_pub.to_csv('donnee/cmd_pub.csv')

## Fichiers cartes


In [7]:
#imports des cartes de la CALL (ville)
carte_CALL =  gpd.read_file('CALL-data/Populations_municipales_CALL_2020.shp')
carte_CALL.info()

<class 'geopandas.geodataframe.GeoDataFrame'>
RangeIndex: 36 entries, 0 to 35
Data columns (total 10 columns):
 #   Column      Non-Null Count  Dtype   
---  ------      --------------  -----   
 0   OBJECTID    36 non-null     int64   
 1   Code_Insee  36 non-null     object  
 2   Libelle     36 non-null     object  
 3   Population  36 non-null     int64   
 4   Evol_annue  36 non-null     float64 
 5   Densite_po  36 non-null     float64 
 6   Populati_1  36 non-null     int64   
 7   Shape__Are  36 non-null     float64 
 8   Shape__Len  36 non-null     float64 
 9   geometry    36 non-null     geometry
dtypes: float64(4), geometry(1), int64(3), object(2)
memory usage: 2.9+ KB


In [8]:
carte_CALL.drop(['OBJECTID','Code_Insee','Population','Evol_annue','Densite_po','Populati_1'],axis=1,inplace=True)
carte_CALL.to_file('donnee/font_carte.gdp')

In [9]:
#import carte des bennes
carte_verre = gpd.read_file('CALL-data/D%C3%A9chets.shp')

In [10]:
carte_verre.info()

<class 'geopandas.geodataframe.GeoDataFrame'>
RangeIndex: 500 entries, 0 to 499
Data columns (total 38 columns):
 #   Column      Non-Null Count  Dtype   
---  ------      --------------  -----   
 0   objectid    500 non-null    int64   
 1   commune     500 non-null    object  
 2   flux        500 non-null    object  
 3   nom         117 non-null    object  
 4   adresse     500 non-null    object  
 5   aériennes   500 non-null    object  
 6   identifian  0 non-null      object  
 7   volume_bor  381 non-null    float64 
 8   marque      498 non-null    object  
 9   modèle      496 non-null    object  
 10  couleur     499 non-null    object  
 11  pédale      500 non-null    object  
 12  trappe_gro  500 non-null    object  
 13  cliiink     458 non-null    object  
 14  etat_borne  126 non-null    object  
 15  surface___  0 non-null      object  
 16  date_de_mi  130 non-null    object  
 17  appartienn  500 non-null    object  
 18  convention  0 non-null      object  
 19  

In [11]:
#je ne garde que la commune et les coordonnées
carte_verre = carte_verre[['commune','geometry']]

In [12]:
carte_verre.to_file('donnee/carteverre.gpd')

<a id=population></a>
## Fichier population 

In [13]:
population = pd.read_csv('CALL-data/Populations_municipales_CALL_2020.csv')

In [14]:
perso_pd.analyseDataFrame(population)

################################# INFO  #################################
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 36 entries, 0 to 35
Data columns (total 9 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   OBJECTID                         36 non-null     int64  
 1   Code_Insee                       36 non-null     int64  
 2   Libelle                          36 non-null     object 
 3   Population_municipale_2020       36 non-null     int64  
 4   Evol_annuelle_moy_population_20  36 non-null     float64
 5   Densite_population_2019_histori  36 non-null     float64
 6   Population_municipale_2019_hist  36 non-null     int64  
 7   Shape__Area                      36 non-null     float64
 8   Shape__Length                    36 non-null     float64
dtypes: float64(4), int64(4), object(1)
memory usage: 2.7+ KB
################################# DESCRIBE  #################################


OBJECTID    Code_Insee               Libelle  \
count   36.000000     36.000000                    36   
unique        NaN           NaN                    36   
top           NaN           NaN  Ablain-Saint-Nazaire   
freq          NaN           NaN                     1   
mean    18.500000  62431.722222                   NaN   
std     10.535654    283.530357                   NaN   
min      1.000000  62001.000000                   NaN   
25%      9.750000  62182.000000                   NaN   
50%     18.500000  62438.500000                   NaN   
75%     27.250000  62637.500000                   NaN   
max     36.000000  62895.000000                   NaN   

        Population_municipale_2020  Evol_annuelle_moy_population_20  \
count                    36.000000                        36.000000   
unique                         NaN                              NaN   
top                            NaN                              NaN   
freq                           NaN                              NaN   
mean                   6728.833333                         0.152778   
std                    7225.996827                         0.748135   
min                     350.000000                        -1.000000   
25%                    2289.750000                        -0.400000   
50%                    4627.000000                        -0.050000   
75%                    8243.500000                         0.600000   
max                   32458.000000                         2.200000   

        Densite_population_2019_histori  Population_municipale_2019_hist  \
count                         36.000000                        36.000000   
unique                              NaN                              NaN   
top                                 NaN                              NaN   
freq                                NaN                              NaN   
mean                        1073.194444                      6701.888889   
std                          835.899147                      7125.903667   
min                           84.000000                       347.000000   
25%                          367.150000                      2288.750000   
50%                          891.250000                      4537.500000   
75%                         1555.375000                      8252.500000   
max                         3017.300000                     31461.000000   

         Shape__Area  Shape__Length  
count   3.600000e+01      36.000000  
unique           NaN            NaN  
top              NaN            NaN  
freq             NaN            NaN  
mean    6.662342e+06   13950.687652  
std     3.481763e+06    4069.467891  
min     1.166704e+06    5549.983908  
25%     3.806048e+06   10510.096444  
50%     5.833812e+06   14023.570172  
75%     1.004338e+07   17865.466128  
max     1.302329e+07   20686.853192

################################# HEAD  #################################


OBJECTID  Code_Insee               Libelle  Population_municipale_2020  \
0         1       62001  Ablain-Saint-Nazaire                        1929   
1         2       62003             Acheville                         599   
2         3       62019          Aix-Noulette                        3917   
3         4       62032                Angres                        4777   
4         5       62033                 Annay                        4477   

   Evol_annuelle_moy_population_20  Densite_population_2019_histori  \
0                              1.4                            190.6   
1                             -1.0                            198.7   
2                             -0.1                            375.8   
3                              2.2                            974.7   
4                              1.1                           1010.9   

   Population_municipale_2019_hist   Shape__Area  Shape__Length  
0                             1877  9.957907e+06   17778.342599  
1                              604  3.073788e+06    9485.464858  
2                             3923  1.044563e+07   20320.061142  
3                             4698  4.835830e+06   14054.685325  
4                             4377  4.328373e+06   10695.642876

################################# DOUBLONS  #################################


Empty DataFrame
Columns: [OBJECTID, Code_Insee, Libelle, Population_municipale_2020, Evol_annuelle_moy_population_20, Densite_population_2019_histori, Population_municipale_2019_hist, Shape__Area, Shape__Length]
Index: []

################################## recherche de clé pour merge ###################
la colonne OBJECTID est une clé
la colonne Code_Insee est une clé
la colonne Libelle est une clé
la colonne Densite_population_2019_histori est une clé
la colonne Population_municipale_2019_hist est une clé
la colonne Shape__Area est une clé
la colonne Shape__Length est une clé


pas de doublons, le fichier est propre. J'enleve que les colonnes Code_Insee et Evol_annuelle_moy_population_20<br>
Je crée aussi un fichier cate pour l'affichage des cartes.<br>

In [15]:
population = population.drop(['Code_Insee','Evol_annuelle_moy_population_20'
                              ,'Shape__Area','Shape__Length'
                              ,'OBJECTID','Densite_population_2019_histori'
                              ,'Population_municipale_2019_hist'
                            ], axis=1)
#remplacement nom des villes car pas les meme avec le fichier des bennes
population.replace({'Annay' : 'Annay-sous-Lens'
                   ,'Éleu-dit-Leauwette' : 'Eleu-dit-Leauwette'
                   ,'Estevelles' :'Estévelles'}
                   ,inplace=True)


In [16]:
population.head(50)

Libelle  Population_municipale_2020
0   Ablain-Saint-Nazaire                        1929
1              Acheville                         599
2           Aix-Noulette                        3917
3                 Angres                        4777
4        Annay-sous-Lens                        4477
5                  Avion                       17676
6           Bénifontaine                         350
7         Billy-Montigny                        8185
8     Bouvigny-Boyeffles                        2376
9        Bully-les-Mines                       12080
10               Carency                         796
11    Eleu-dit-Leauwette                        2844
12            Estévelles                        2031
13   Fouquières-lès-Lens                        6333
14   Givenchy-en-Gohelle                        2003
15          Gouy-Servins                         350
16                Grenay                        6748
17                Harnes                       12277
18               Hulluch                        3380
19                  Lens                       32458
20                Liévin                       30102
21      Loison-sous-Lens                        5308
22       Loos-en-Gohelle                        6904
23            Mazingarbe                        8029
24             Méricourt                       11374
25              Meurchin                        3728
26    Noyelles-sous-Lens                        6590
27         Pont-à-Vendin                        3116
28      Sains-en-Gohelle                        6035
29           Sallaumines                        9575
30               Servins                        1133
31               Souchez                        2643
32       Vendin-le-Vieil                        8419
33       Villers-au-Bois                         614
34                  Vimy                        4332
35               Wingles                        8750

<a id=verre></a>


## Fichier déchets

In [17]:
benne_verre = pd.read_csv('CALL-data/Déchets.csv')

In [18]:
perso_pd.analyseDataFrame(benne_verre)

################################# INFO  #################################
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 500 entries, 0 to 499
Data columns (total 39 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   X                                498 non-null    float64
 1   Y                                498 non-null    float64
 2   objectid                         500 non-null    int64  
 3   commune                          500 non-null    object 
 4   flux                             500 non-null    object 
 5   nom                              117 non-null    object 
 6   adresse                          500 non-null    object 
 7   aériennes__ba___enterrées__be__  500 non-null    object 
 8   identifiant                      0 non-null      float64
 9   volume_borne__m3_                381 non-null    float64
 10  marque                           498 non-null    object 
 11  modèle    

X           Y    objectid commune   flux  \
count   498.000000  498.000000  500.000000     500    500   
unique         NaN         NaN         NaN      37      1   
top            NaN         NaN         NaN    Lens  Verre   
freq           NaN         NaN         NaN     113    500   
mean      2.813075   50.433711  250.500000     NaN    NaN   
std       0.058954    0.023062  144.481833     NaN    NaN   
min       2.633134   50.367027    1.000000     NaN    NaN   
25%       2.769478   50.418942  125.750000     NaN    NaN   
50%       2.817495   50.433411  250.500000     NaN    NaN   
75%       2.848384   50.445672  375.250000     NaN    NaN   
max       2.925140   50.500057  500.000000     NaN    NaN   

                                  nom  \
count                             117   
unique                             80   
top     Résidence Sellier - Cité du 4   
freq                                5   
mean                              NaN   
std                               NaN   
min                               NaN   
25%                               NaN   
50%                               NaN   
75%                               NaN   
max                               NaN   

                                                 adresse  \
count                                                500   
unique                                               484   
top     Intersection rues Louis Montcalm et Jules Guesde   
freq                                                   2   
mean                                                 NaN   
std                                                  NaN   
min                                                  NaN   
25%                                                  NaN   
50%                                                  NaN   
75%                                                  NaN   
max                                                  NaN   

       aériennes__ba___enterrées__be__  identifiant  volume_borne__m3_  ...  \
count                              500          0.0         381.000000  ...   
unique                               3          NaN                NaN  ...   
top                                 BA          NaN                NaN  ...   
freq                               373          NaN                NaN  ...   
mean                               NaN          NaN           3.992126  ...   
std                                NaN          NaN           0.088502  ...   
min                                NaN          NaN           3.000000  ...   
25%                                NaN          NaN           4.000000  ...   
50%                                NaN          NaN           4.000000  ...   
75%                                NaN          NaN           4.000000  ...   
max                                NaN          NaN           4.000000  ...   

             lon      lon2 dimensions_des_plaques_de_consi  \
count        498       498                             0.0   
unique       486       486                             NaN   
top     2,756227  2,756227                             NaN   
freq           2         2                             NaN   
mean         NaN       NaN                             NaN   
std          NaN       NaN                             NaN   
min          NaN       NaN                             NaN   
25%          NaN       NaN                             NaN   
50%          NaN       NaN                             NaN   
75%          NaN       NaN                             NaN   
max          NaN       NaN                             NaN   

       consignes_de_tri_à_jour        lat3        lon3 created_user  \
count                      272  498.000000  498.000000          500   
unique                       3         NaN         NaN            1   
top                        Non         NaN         NaN    AdminAgso   
freq                       136         NaN         NaN          500   
mean                 

################################# HEAD  #################################


X          Y  objectid               commune   flux  nom  \
0  2.712467  50.393250         1  Ablain-Saint-Nazaire  Verre  NaN   
1  2.712402  50.390385         2  Ablain-Saint-Nazaire  Verre  NaN   
2  2.709638  50.392372         3  Ablain-Saint-Nazaire  Verre  NaN   
3  2.888295  50.385292         4             Acheville  Verre  NaN   
4  2.712122  50.421780         5          Aix-Noulette  Verre  NaN   

                                             adresse  \
0                                  Allée Jules Ferry   
1  Intersection rue Gares et résidence des Sœurs ...   
2  Rue Marcel Lancino - Proche de l'intersection ...   
3  Rue du Maréchal Leclerc - A la sortie de la vi...   
4  Allée des Bleuets - Sur le parking du stade Ga...   

  aériennes__ba___enterrées__be__  identifiant  volume_borne__m3_  ...  \
0                              BA          NaN                4.0  ...   
1                              BA          NaN                4.0  ...   
2                              BA          NaN                4.0  ...   
3                              BA          NaN                4.0  ...   
4                              BA          NaN                4.0  ...   

        lon      lon2 dimensions_des_plaques_de_consi consignes_de_tri_à_jour  \
0  2,712467  2,712467                             NaN                     NaN   
1  2,712402  2,712402                             NaN                     NaN   
2  2,709638  2,709638                             NaN                     Oui   
3  2,888295  2,888295                             NaN                     Non   
4  2,712122  2,712122                             NaN                     NaN   

        lat3      lon3 created_user                created_date  \
0  50.393250  2.712467    AdminAgso  2022/01/31 08:36:17.351+00   
1  50.390385  2.712402    AdminAgso  2022/01/31 08:36:17.351+00   
2  50.392372  2.709638    AdminAgso  2022/01/31 08:36:17.351+00   
3  50.385292  2.888295    AdminAgso  2022/01/31 08:36:17.351+00   
4  50.421780  2.712122    AdminAgso  2022/01/31 08:36:17.351+00   

  last_edited_user            last_edited_date  
0        AdminAgso  2022/01/31 08:36:17.351+00  
1        AdminAgso  2022/01/31 08:36:17.351+00  
2        AdminAgso  2022/01/31 08:36:17.351+00  
3        AdminAgso  2022/01/31 08:36:17.351+00  
4        AdminAgso  2022/01/31 08:36:17.351+00  

[5 rows x 39 columns]

################################# DOUBLONS  #################################


Empty DataFrame
Columns: [X, Y, objectid, commune, flux, nom, adresse, aériennes__ba___enterrées__be__, identifiant, volume_borne__m3_, marque, modèle, couleur, pédale, trappe_gros_producteurs, cliiink, etat_borne, surface___terre__ciment__herbe_, date_de_mise_en_service, appartiennent_à, convention, date_fin_convention, informations, nombre_de_logements_collectifs, nombre_de_logements_individuels, relevé_gps_code_1, lat, lat2, relevé_gps_code_2, lon, lon2, dimensions_des_plaques_de_consi, consignes_de_tri_à_jour, lat3, lon3, created_user, created_date, last_edited_user, last_edited_date]
Index: []

[0 rows x 39 columns]

################################## recherche de clé pour merge ###################
la colonne objectid est une clé


In [19]:
benne_verre = benne_verre.drop(['objectid'
                                ], axis=1)

In [20]:
display(benne_verre[benne_verre.duplicated()])

X          Y             commune   flux                     nom  \
95   2.678842  50.437225  Bouvigny-Boyeffles  Verre                     NaN   
253  2.829325  50.436352                Lens  Verre  Résidence Luminésens 2   
372  2.780912  50.433556              Liévin  Verre                     NaN   
398  2.866120  50.445709    Loison-sous-Lens  Verre                     NaN   
434  2.756227  50.468002          Mazingarbe  Verre                     NaN   
439  2.868818  50.405617           Méricourt  Verre                     NaN   
493  2.872612  50.496155             Wingles  Verre     Cité de la Verrerie   

                                               adresse  \
95        Intersection rue du Prince et chemin Latéral   
253       Intersection rues de Londres et Angela Davis   
372   Intersection rues Louis Montcalm et Jules Guesde   
398  Intersection rue de l'Abbaye et chemin de Lois...   
434              Route Nationale - A côté du numéro 91   
439   Avenue de Flöha - A côté des services techniques   
493       Intersection rues des Erables et des Charmes   

    aériennes__ba___enterrées__be__  identifiant  volume_borne__m3_  marque  \
95                               BA          NaN                4.0    Sulo   
253                              BE          NaN                NaN  Tercol   
372                              BA          NaN                4.0    UTPM   
398                              BA          NaN                4.0    UTPM   
434                              BA          NaN                4.0    UTPM   
439                              BA          NaN                4.0    UTPM   
493                              BE          NaN                NaN    Sulo   

     ...       lon      lon2 dimensions_des_plaques_de_consi  \
95   ...  2,678842  2,678842                             NaN   
253  ...  2,829325  2,829325                             NaN   
372  ...  2,780912  2,780912                             NaN   
398  ...  2,866120  2,866120                             NaN   
434  ...  2,756227  2,756227                             NaN   
439  ...  2,868818  2,868818                             NaN   
493  ...  2,872612  2,872612                             NaN   

    consignes_de_tri_à_jour       lat3      lon3  created_user  \
95                      Non  50.437225  2.678842     AdminAgso   
253       Oui (autocollant)  50.436352  2.829325     AdminAgso   
372                     Oui  50.433556  2.780912     AdminAgso   
398                     Oui  50.445709  2.866120     AdminAgso   
434                     Oui  50.468002  2.756227     AdminAgso   
439                     Non  50.405617  2.868818     AdminAgso   
493                     Non  50.496155  2.872612     AdminAgso   

                   created_date last_edited_user            last_edited_date  
95   2022/01/31 08:36:17.351+00        AdminAgso  2022/01/31 08:36:17.351+00  
253  2022/01/31 08:36:17.351+00        AdminAgso  2022/01/31 08:36:17.351+00  
372  2022/01/31 08:36:17.351+00        AdminAgso  2022/01/31 08:36:17.351+00  
398  2022/01/31 08:36:17.351+00        AdminAgso  2022/01/31 08:36:17.351+00  
434  2022/01/31 08:36:17.351+00        AdminAgso  2022/01/31 08:36:17.351+00  
439  2022/01/31 08:36:17.351+00        AdminAgso  2022/01/31 08:36:17.351+00  
493  2022/01/31 08:36:17.351+00        AdminAgso  2022/01/31 08:36:17.351+00  

[7 rows x 38 columns]

Il y a réelement des bennes en double sur certain site(il y a 2 bennes) d'oú les doublons.<br>
Je vais supprimer les colonnes inutiles.<br>

In [21]:
benne_verre = benne_verre.drop(['flux','created_user','created_date','last_edited_user','last_edited_date'
                                ,'identifiant','pédale','lon','lon2','lon3','lat','lat2','lat3'
                                ,'dimensions_des_plaques_de_consi','relevé_gps_code_1','relevé_gps_code_2'
                                ,'nombre_de_logements_individuels','nombre_de_logements_collectifs'
                                ,'surface___terre__ciment__herbe_','convention','date_fin_convention'
                                ,'informations','nom','adresse','marque','modèle','couleur','etat_borne'
                                ,'date_de_mise_en_service','appartiennent_à','aériennes__ba___enterrées__be__'
                                ,'X','Y'
                                ], axis=1)

In [22]:
#objectif : une ligne par ville avec : le nombre de benne ,le pourcentage de benne enterré
#,le volume total,le pourcentage de trappe gros producteur, le pourcentage Cliiink
#,pourcentage avec des consigne a jour
benne_verre.replace({'Annay' : 'Annay-sous-Lens'
                   ,'Éleu-dit-Leauwette' : 'Eleu-dit-Leauwette'
                   ,'Estevelles' :'Estévelles'}
                   ,inplace=True)


In [23]:
#transformer les Oui en 1
benne_verre['consignes_de_tri_à_jour'].fillna(0,inplace=True)
benne_verre['trappe_gros_producteurs'] = benne_verre['trappe_gros_producteurs'].apply(lambda x: 1 if x =='Oui' else 0)
benne_verre['cliiink'] = benne_verre['cliiink'].apply(lambda x: 1 if x =='Oui' else 0)
benne_verre['consignes_de_tri_à_jour'] = benne_verre['consignes_de_tri_à_jour'].apply(lambda x: 1 if x =='Oui' else 0)


In [24]:
#faire une colonne nombre de bennes
benne_verre['nb_benne']=1

In [25]:
#regrouper les communes et faire la somme
benne_verre = benne_verre.groupby('commune').sum()
benne_verre.reset_index(inplace=True)

In [26]:
fichier_final_dechet = benne_verre.merge(population,left_on='commune',right_on='Libelle',how='outer')
fichier_final_dechet.drop(['Libelle'],axis=1,inplace=True)

In [27]:
fichier_final_dechet.to_csv('donnee/benne_verre.csv',index=False)

<a id=ecole ></a>


## Fichier établissement scolaire

In [28]:
etb_scolaire = pd.read_csv('CALL-data/Etablissements_Enseignement.csv')

In [29]:
perso_pd.analyseDataFrame(etb_scolaire)

################################# INFO  #################################
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 236 entries, 0 to 235
Data columns (total 41 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   X           236 non-null    float64
 1   Y           236 non-null    float64
 2   patronyme_  236 non-null    object 
 3   objectid    236 non-null    int64  
 4   numero_uai  236 non-null    object 
 5   appellatio  236 non-null    object 
 6   denominati  236 non-null    object 
 7   secteur_pu  236 non-null    object 
 8   adresse_ua  236 non-null    object 
 9   lieu_dit_u  231 non-null    object 
 10  boite_post  231 non-null    object 
 11  code_posta  236 non-null    int64  
 12  localite_a  236 non-null    object 
 13  libelle_co  236 non-null    object 
 14  coordonnee  231 non-null    float64
 15  coordonn_1  231 non-null    float64
 16  epsg        231 non-null    object 
 17  latitude    231 non-null    float64
 

X           Y   patronyme_    objectid numero_uai  \
count   236.000000  236.000000          236  236.000000        236   
unique         NaN         NaN          165         NaN        236   
top            NaN         NaN  JEAN JAURES         NaN   0620108F   
freq           NaN         NaN            7         NaN          1   
mean      2.811179   50.433146          NaN  118.500000        NaN   
std       0.063584    0.025969          NaN   68.271517        NaN   
min       2.639194   50.370849          NaN    1.000000        NaN   
25%       2.765603   50.416034          NaN   59.750000        NaN   
50%       2.817319   50.429847          NaN  118.500000        NaN   
75%       2.862953   50.446271          NaN  177.250000        NaN   
max       2.921097   50.495932          NaN  236.000000        NaN   

                        appellatio               denominati secteur_pu  \
count                          236                      236        236   
unique                         196                       21          2   
top     Ecole primaire Jean Jaurès  ECOLE PRIMAIRE PUBLIQUE     Public   
freq                             4                       89        225   
mean                           NaN                      NaN        NaN   
std                            NaN                      NaN        NaN   
min                            NaN                      NaN        NaN   
25%                            NaN                      NaN        NaN   
50%                            NaN                      NaN        NaN   
75%                            NaN                      NaN        NaN   
max                            NaN                      NaN        NaN   

                   adresse_ua lieu_dit_u  ... libelle_ac  secteur_pr  \
count                     236        231  ...        236  236.000000   
unique                    218         11  ...          1         NaN   
top     Rue FRANCOIS DELATTRE             ...      Lille         NaN   
freq                        2        221  ...        236         NaN   
mean                      NaN        NaN  ...        NaN   95.737288   
std                       NaN        NaN  ...        NaN   15.674578   
min                       NaN        NaN  ...        NaN   10.000000   
25%                       NaN        NaN  ...        NaN   99.000000   
50%                       NaN        NaN  ...        NaN   99.000000   
75%                       NaN        NaN  ...        NaN   99.000000   
max                       NaN        NaN  ...        NaN   99.000000   

        secteur__1     eff_1920  statut                dat_ferm euroscol  \
count          236   236.000000     236                       6      236   
unique           3          NaN       2                       3        2   
top     SANS OBJET          NaN  Ouvert  2019/08/31 00:00:00+00      NON   
freq           226          NaN     230                       3      234   
mean           NaN   209.012712     NaN                     NaN      NaN   
std            NaN   207.175277     NaN                     NaN      NaN   
min            NaN     0.000000     NaN                     NaN      NaN   
25%            NaN    96.500000     NaN                     NaN      NaN   
50%            NaN   148.500000     NaN                     NaN      NaN   
75%            NaN   235.500000     NaN                     NaN      NaN   
max            NaN  1188.000000     NaN                     NaN      NaN   

           eff_2021     eff_2122 eff_2223  
count    236.000000   236.000000      0.0  
unique          NaN          NaN      NaN  
top             NaN          NaN      NaN  
freq            NaN          NaN      NaN  
mean     206.800847   205.097458      NaN  
std      207.413470   209.845544      NaN  
min        0.000000     0.000000      NaN  
25%       91.750000    89.750000      NaN  
50%      140.000000   141.000000      NaN  
75%      229.250000   234.000000      NaN  
max     1203.000000  1256

################################# HEAD  #################################


X          Y        patronyme_  objectid numero_uai  \
0  2.838308  50.431341         CONDORCET         1   0620108F   
1  2.739299  50.441818  EMILIENNE MOREAU         2   0620276N   
2  2.844518  50.436474            ROLAND         3   0620302S   
3  2.777397  50.426335      LEO LAGRANGE         4   0620309Z   
4  2.845627  50.491154       AURIOL-JOLY         5   0620376X   

                                 appellatio                      denominati  \
0  Lycée général et technologique Condorcet  LYCEE GENERAL ET TECHNOLOGIQUE   
1         Ecole maternelle Emilienne Moreau       ECOLE MATERNELLE PUBLIQUE   
2                   Ecole maternelle Roland       ECOLE MATERNELLE PUBLIQUE   
3             Ecole maternelle Léo Lagrange       ECOLE MATERNELLE PUBLIQUE   
4           Ecole maternelle Vincent Auriol       ECOLE MATERNELLE PUBLIQUE   

  secteur_pu                    adresse_ua lieu_dit_u  ... libelle_ac  \
0     Public         25  RUE ETIENNE DOLET             ...      Lille   
1     Public                Rue de Marengo             ...      Lille   
2     Public      20 A BOULEVARD DU MARAIS             ...      Lille   
3     Public               Rue Montgolfier             ...      Lille   
4     Public  1 BIS RUE DES FRERES BUISINE             ...      Lille   

   secteur_pr  secteur__1 eff_1920  statut  dat_ferm euroscol  eff_2021  \
0          99  SANS OBJET      682  Ouvert       NaN      NON       681   
1          99  SANS OBJET      142  Ouvert       NaN      NON       124   
2          99  SANS OBJET       76  Ouvert       NaN      NON        78   
3          99  SANS OBJET       97  Ouvert       NaN      NON        89   
4          99  SANS OBJET      201  Ouvert       NaN      NON       200   

   eff_2122 eff_2223  
0       701      NaN  
1       124      NaN  
2        80      NaN  
3        92      NaN  
4       188      NaN  

[5 rows x 41 columns]

################################# DOUBLONS  #################################


Empty DataFrame
Columns: [X, Y, patronyme_, objectid, numero_uai, appellatio, denominati, secteur_pu, adresse_ua, lieu_dit_u, boite_post, code_posta, localite_a, libelle_co, coordonnee, coordonn_1, epsg, latitude, longitude, appariemen, localisati, nature_uai, nature_u_1, etat_etabl, etat_eta_1, code_depar, code_regio, code_acade, code_commu, libelle_de, libelle_re, libelle_ac, secteur_pr, secteur__1, eff_1920, statut, dat_ferm, euroscol, eff_2021, eff_2122, eff_2223]
Index: []

[0 rows x 41 columns]

################################## recherche de clé pour merge ###################
la colonne objectid est une clé
la colonne numero_uai est une clé


<a id=numerique ></a>

## Fichier établissement numérique

In [30]:
etb_num = pd.read_csv('CALL-data/Lieux_et_acteurs_de_la_médiation_numérique.csv')

In [31]:
perso_pd.analyseDataFrame(etb_num)

################################# INFO  #################################
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 55 entries, 0 to 54
Data columns (total 64 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   X                                55 non-null     float64
 1   Y                                55 non-null     float64
 2   objectid                         55 non-null     int64  
 3   id                               54 non-null     object 
 4   date_maj                         55 non-null     object 
 5   nom                              55 non-null     object 
 6   statut                           54 non-null     object 
 7   epn                              55 non-null     object 
 8   nature                           54 non-null     object 
 9   horaires_ouverture               54 non-null     object 
 10  email_contact                    54 non-null     object 
 11  adresse     

X          Y  objectid    id                date_maj  \
count   55.000000  55.000000  55.00000    54                      55   
unique        NaN        NaN       NaN    54                      19   
top           NaN        NaN       NaN  1569  2018/11/01 00:00:00+00   
freq          NaN        NaN       NaN     1                      16   
mean     2.821023  50.433117  28.00000   NaN                     NaN   
std      0.063460   0.022228  16.02082   NaN                     NaN   
min      2.683968  50.395073   1.00000   NaN                     NaN   
25%      2.770453  50.420146  14.50000   NaN                     NaN   
50%      2.831664  50.430153  28.00000   NaN                     NaN   
75%      2.872900  50.444285  41.50000   NaN                     NaN   
max      2.917293  50.495318  55.00000   NaN                     NaN   

                                      nom  statut  epn  \
count                                  55      54   55   
unique                                 51       2    2   
top     Accompagnement numérique - Mairie  Public  Non   
freq                                    5      46   45   
mean                                  NaN     NaN  NaN   
std                                   NaN     NaN  NaN   
min                                   NaN     NaN  NaN   
25%                                   NaN     NaN  NaN   
50%                                   NaN     NaN  NaN   
75%                                   NaN     NaN  NaN   
max                                   NaN     NaN  NaN   

                            nature horaires_ouverture  ... itinérance  \
count                           54                 54  ...         54   
unique                          10                 16  ...          1   
top     Bibliothèque - Médiathèque              -----  ...      -----   
freq                            14                 37  ...         54   
mean                           NaN                NaN  ...        NaN   
std                            NaN                NaN  ...        NaN   
min                            NaN                NaN  ...        NaN   
25%                            NaN                NaN  ...        NaN   
50%                            NaN                NaN  ...        NaN   
75%                            NaN                NaN  ...        NaN   
max                            NaN                NaN  ...        NaN   

       france_services    département arrondissement                 epci  \
count               54             54             55                   55   
unique               3              1              1                    1   
top              -----  Pas-de-Calais           Lens  CA de Lens - Liévin   
freq                46             54             55                   55   
mean               NaN            NaN            NaN                  NaN   
std                NaN            NaN            NaN                  NaN   
min                NaN            NaN            NaN                  NaN   
25%                NaN            NaN            NaN                  NaN   
50%                NaN            NaN            NaN                  NaN   
75%                NaN            NaN            NaN                  NaN   
max                NaN            NaN            NaN                  NaN   

       created_user                created_date last_edited_user  \
count            55                          55               55   
unique            1                           1                1   
top       AdminAgso  2022/07/26 09:04:59.639+00        AdminAgso   
freq             55                          55               55   
mean            NaN                         NaN              NaN   
std             NaN                         NaN              NaN   
min             NaN                         NaN              NaN   
25%             NaN                         NaN              NaN   
50%             NaN                     

################################# HEAD  #################################


X          Y  objectid    id                date_maj  \
0  2.759208  50.406966         1  1569  2021/07/20 00:00:00+00   
1  2.763153  50.406796         2  2128  2021/03/26 00:00:00+00   
2  2.877767  50.463210         3  0667  2018/11/01 00:00:00+00   
3  2.877154  50.462582         4  1804  2020/11/27 00:00:00+00   
4  2.834432  50.412593         5  0678  2018/11/01 00:00:00+00   

                                  nom  statut  epn  \
0  Médiathèque L'Embellie / cyberbase  Public  Non   
1                   Le moulin de tous  Public  Non   
2             Bibliotheque Municipale  Public  Non   
3   Accompagnement numérique - Mairie  Public  Non   
4              Médiathèque Emile Zola  Public  Non   

                                   nature horaires_ouverture  ... itinérance  \
0              Bibliothèque - Médiathèque              -----  ...      -----   
1  Collectivité territoriale, Ville, CCAS              -----  ...      -----   
2              Bibliothèque - Médiathèque              -----  ...      -----   
3  Collectivité territoriale, Ville, CCAS              -----  ...      -----   
4              Bibliothèque - Médiathèque              -----  ...      -----   

  france_services    département arrondissement                 epci  \
0             Non  Pas-de-Calais           Lens  CA de Lens - Liévin   
1           -----  Pas-de-Calais           Lens  CA de Lens - Liévin   
2           -----  Pas-de-Calais           Lens  CA de Lens - Liévin   
3           -----  Pas-de-Calais           Lens  CA de Lens - Liévin   
4           -----  Pas-de-Calais           Lens  CA de Lens - Liévin   

  created_user                created_date last_edited_user  \
0    AdminAgso  2022/07/26 09:04:59.639+00        AdminAgso   
1    AdminAgso  2022/07/26 09:04:59.639+00        AdminAgso   
2    AdminAgso  2022/07/26 09:04:59.639+00        AdminAgso   
3    AdminAgso  2022/07/26 09:04:59.639+00        AdminAgso   
4    AdminAgso  2022/07/26 09:04:59.639+00        AdminAgso   

             last_edited_date                              globalid  
0  2022/07/26 09:04:59.639+00  37480280-a29d-49ac-abbe-1790f8ba5955  
1  2022/07/26 09:04:59.639+00  0d76c7be-79ce-48c4-b8f5-220f4cf47ae7  
2  2022/07/26 09:04:59.639+00  ec48508f-b8e8-43bf-8dc7-43477e292b30  
3  2022/07/26 09:04:59.639+00  30f67524-e91a-4ce4-be18-f7f597639990  
4  2022/07/26 09:04:59.639+00  70945d72-d321-4a54-92a0-ab146906df05  

[5 rows x 64 columns]

################################# DOUBLONS  #################################


Empty DataFrame
Columns: [X, Y, objectid, id, date_maj, nom, statut, epn, nature, horaires_ouverture, email_contact, adresse, code_postal, commune, téléphone, site_internet, accès_équipement, type_équipement, conditions_accès_équipement, coût_accès_équipement, horaires_équipement, médiation_numérique, type_médiation_numérique, conditions_accès_médiation_numé, accompagnement_médiation_numéri, coût_accès_médiation_numérique, fréquence_médiation_numérique, horaires_médnum, démarches, démarches_spécifiques, type_démarches, conditions_accès_démarches, accompagnement_démarches, coût_accès_démarches, fréquence_démarches, horaires_démarches, stockage, coût_stockage, vente_matériel, type_vente_matériel, détail_publics, détail_territoire, accessibilité_pmr, accès_transports_en_commun, aptic, latitude, longitude, horaires_lundi, horaires_mardi, horaires_mercredi, horaires_jeudi, horaires_vendredi, horaires_samedi, horaires_dimanche, itinérance, france_services, département, arrondissement, epci, created_user, created_date, last_edited_user, last_edited_date, globalid]
Index: []

[0 rows x 64 columns]

################################## recherche de clé pour merge ###################
la colonne X est une clé
la colonne Y est une clé
la colonne objectid est une clé
la colonne id est une clé
la colonne adresse est une clé
la colonne latitude est une clé
la colonne longitude est une clé
la colonne globalid est une clé
